File: 2020/python nov 2024/day 2/day2.py

def main():
    part_one("../textfile/day2final.txt")
    part_two("../textfile/day2final.txt")

def part_one(file_name:str):
    with open(file_name, encoding="utf-8") as f:
        cnt = 0
        for line in f.readlines():
            split = line.split(" ")
            low,high = [int(x) for x in  split[0].split("-")]
            count_of_letters = split[2].count(split[1][0])
            if count_of_letters >= low and count_of_letters <= high:
                cnt+=1
        print("Day two part one:",cnt)

def part_two(file_name:str):
    with open(file_name, encoding="utf-8") as f:
        cnt = 0
        for line in f.readlines():
            split = line.split(" ")
            low,high = [int(x) for x in  split[0].split("-")]
            c = split[1][0]

            if (split[2][low-1] == c and split[2][high-1] != c) or  (split[2][low-1] != c and split[2][high-1] == c):
                cnt+=1

        print("Day 2 part two:",cnt)


main()

File: 2019/CPP Feb 2025/Day 2/day2.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>

#include "day2.h"

int main()
{
    std::string fileName{"../textFiles/day2final.txt"};
    partOne(fileName);
}

void partOne(std::string fileName)
{
    std::ifstream file{fileName};

    if (file.is_open())
    {
        std::string line{};
        std::vector<int> codes{};
        std::string tok{};
        while (std::getline(file, line))
        {
            std::stringstream ss{line};

            while (std::getline(ss, tok, ','))
            {
                codes.push_back(std::stoi(tok));
            }
        }
        int currentPos{0};
        //make sure to read the whole problem statement
        codes[1] = 12;
        codes[2] = 2;

        while (codes[currentPos] != 99)
        {
            int a{codes[codes[currentPos + 1]]};
            int b{codes[codes[currentPos + 2]]};
            if (codes[currentPos] == 1)
            {
                codes[codes[currentPos + 3]] = a + b;
            }
            else if (codes[currentPos] == 2)
            {
                codes[codes[currentPos + 3]] = a * b;
            }
            currentPos += 4;
        }
        std::cout << "day 2 part one: " << codes[0] << std::endl;
    }
    else
    {
        std::cout << "file won't open" << std::endl;
    }
}